pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -190,9 +190,15 @@ public synchronized CompletableFuture<Void> addConsumer(Consumer consumer) {
         }
 
         if (isConsumersExceededOnSubscription()) {
-            log.warn("[{}] Attempting to add consumer to subscription which reached max consumers limit", name);
+            log.warn("[{}] Attempting to add consumer to subscription which reached max consumers limit {}",
+                    name, consumer);
             return FutureUtil.failedFuture(new ConsumerBusyException("Subscription reached max consumers limit"));
         }
+        if (consumerSet.contains(consumer)) {
```

Comment:
Should we add the check to the `NonPersistentDispatcherMultipleConsumers` as well?

Reply:
It should not block this PR because the race is hard to happen for non-persistent consumers. Feel free to resolve this conversation if necessary.

Reply:
Used a new solution that prevents subsequent subscribing until the previous one is done, which guarantees that no race condition about twice subscribing.

Reply:
@poorbarcode Pease help write some comments about this change to say this is not expected. And I think in this case, close the dispatcher or topic might a better solution, just to let the consumers to reconnect again, because we don't think what exactly happened at that time.

Reply:
Added